src/utils/validationFunctions.js

```diff
@@ -18,3 +18,15 @@ export function register(email, password) {
     throw Error(ERR.PASSWORD_REQUIREMENTS)
   }
 }
+
+export function validateCanPatch(req, next) {
+  const { role } = req.user
+  const { id } = req.body
+  const targetUserId = Number(req.params.id)
+  const isUser = id === targetUserId
+  const isTeacher = role === 'TEACHER'
+  if (!isTeacher && !isUser) {
+    throw Error(ERR.NOT_AUTHORISED)
+  }
+  next()
```

Comment:
this is written as a middleware. It should instead just be a function can be called from the controller.